### PR TITLE
fixed all rebocop erros and improved readme file

### PR DESCRIPTION
**Pull request by RoyNyaga**

### CODE REVIEW

File: lib/iterator_methods.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def multiply_els(array)
   array.my_inject { |elem, n| elem + n }
```
Comment (bolah2009) [inline lib/iterator_methods.rb]: #### `REQUIRED`

This method seems not to be tested. Other cases that are not tested is referenced in this issue(#2).

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -1,143 +1,145 @@
...
-            expect(sum).to eq(15)
-        end
-    end
-end
\ No newline at end of file
+  describe '#my_each' do
+    it 'iterates through numbers in an array' do
+      arr = [1, 2, 3, 4, 5]
+      count = 0
+      arr.my_each { |_i| count += 1 }
```
Comment (bolah2009) [inline spec/enumerable_spec.rb]: #### `REQUIRED`

Consider deleting unused argument.

```suggestion
      arr.my_each { count += 1 }
```

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -1,143 +1,145 @@
...
+    it ' iterates and compare if any element in the array have the condition in the block' do
+      arr = %w[word school bus eat]
+      cond = arr.my_any? { |elem| elem == 'take' }
+      expect(cond).to be false
+    end
+  end
+
+  describe '#my_none?' do
+    it 'returns true' do
+      arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
```
Comment (bolah2009) [inline spec/enumerable_spec.rb]: #### `REQUIRED`

Consider using `let` instead, this file have different places `arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]` was defined.
Learn more [here](https://github.com/rubocop-hq/rspec-style-guide#let-blocks)

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -1,143 +1,145 @@
...
+    end
+
+    it 'iterates and returns string of lenght 5' do
+      arr2 = %w[how there many carry taken yes]
+      arr_select2 = arr2.my_select { |word| word.length == 5 }
+      expect(arr_select2).to match_array(%w[there carry taken])
+    end
+  end
+
+  describe ' #my_all?' do
```
Comment (bolah2009) [inline spec/enumerable_spec.rb:50]: #### `REQUIRED`

Add test for cases where `block` is not given. See issue #2.

File: spec/enumerable_spec.rb [hunk 5]
```diff
@@ -1,143 +1,145 @@
...
+    end
+
+    it ' returns true when all elements start with s' do
+      arr = %w[school soccer sugar rain]
+      cond = arr.my_all? { |elem| elem[0] == 's' }
+      expect(cond).to be false
+    end
+  end
+
+  describe ' #my_any?' do
```
Comment (bolah2009) [inline spec/enumerable_spec.rb:77]: #### `REQUIRED`

Add test for cases where `block` is not given. See issue #2.

File: spec/enumerable_spec.rb [hunk 6]
```diff
@@ -1,143 +1,145 @@
...
+      arr = %w[name lastname age]
+      cond = arr.my_none? { |elem| elem == 'name' }
+      expect(cond).to be false
+    end
+  end
+
+  describe ' #my_count' do
+    it ' without arguments nor block' do
+      arr = [1, 2, 3, 4]
+      expect(arr.length).to eql(arr.my_count)
```
Comment (bolah2009) [inline spec/enumerable_spec.rb]: #### `REQUIRED`

Add the method to be tested in the `expect` parameter.

```ruby
   expect(actual).to be(expected) # passes if actual.equal?(expected)
```

Learn more [here](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers)

```suggestion
      expect(arr.my_count).to eql(arr.length)
```

File: spec/enumerable_spec.rb [hunk 7]
```diff
@@ -1,143 +1,145 @@
...
+    end
+
+    it ' with a block and argument' do
+      count = [1, 3, 3, 4].my_count(3) { |num| num > 1 }
+      expect(count).to eql(2)
+    end
+
+    it 'comparing count and my_count methods' do
+      arr = [1, 2, 3, 4]
+      expect(arr.count).to eql(arr.my_count)
```
Comment (bolah2009) [inline spec/enumerable_spec.rb]: #### `REQUIRED`

Add the method to be tested in the `expect` parameter.

```ruby
   expect(actual).to be(expected) # passes if actual.equal?(expected)
```

Learn more [here](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers)

```suggestion
      expect(arr.my_count).to eql(arr.count)
```

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Awesome Job Team :clap: 

Check out the review comments and issue (#2). :blush: 

Fix this and submit another code review request. :ok_hand: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Comment by RoyNyaga** [comment]
Thanks for the review sir will fix those today

File: lib/iterator_methods.rb (hunk 1 above)
Comment (AnthonyTC89) [reply lib/iterator_methods.rb]: done!

File: spec/enumerable_spec.rb (hunk 4 above)
Comment (AnthonyTC89) [reply spec/enumerable_spec.rb:50]: done!

File: spec/enumerable_spec.rb (hunk 5 above)
Comment (AnthonyTC89) [reply spec/enumerable_spec.rb:77]: done!

File: spec/enumerable_spec.rb (hunk 6 above)
Comment (AnthonyTC89) [reply spec/enumerable_spec.rb]: nice view!.. done!

File: spec/enumerable_spec.rb (hunk 7 above)
Comment (AnthonyTC89) [reply spec/enumerable_spec.rb]: done!

**Review by Ispirett — APPROVED** [review]
Greate job!!!! @RoyNyaga @AnthonyTC89  :1st_place_medal:   
This project looks good to me you can merge.

##### Before merge 
please delete commentied code in you ` iterator_methods1

#### Optional not required to complete this project.
consider using this flag in you `.rspec` file `--format documentation`

**Pull request merged**